lib/internal/modules/esm/import_map.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+const { isURL, URL } = require('internal/url');
+const { ObjectEntries, ObjectKeys, SafeMap, ArrayIsArray } = primordials;
+const { codes: { ERR_INVALID_IMPORT_MAP } } = require('internal/errors');
+
+class ImportMap {
+  #baseURL;
+  imports = new SafeMap();
+  scopes = new SafeMap();
+
+  constructor(raw, baseURL) {
+    this.#baseURL = baseURL;
+    processImportMap(this, this.#baseURL, raw);
+  }
+
+  get baseURL() {
+    return this.#baseURL;
+  }
+
+  resolve(specifier, parentURL = this.baseURL) {
```

Comment:
```suggestion
  resolve(specifier, parentURL = this.#baseURL) {
```
no need to use the getter here

Reply:
Ah, reading this again and I was thinking about making `baseURL` just a normal property. I was thinking that maybe a loader might want to actually modify this?

Reply:
Keeping it private for now. It it better to open it up later than start with it open.